### PR TITLE
Javascript javascript1 week2

### PR DESCRIPTION
**Pull request by RemyaManoharan**
Finished my homework

### CODE REVIEW

File: javascript/javascript1/week2/weather.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+function clothesBaseTemperature(temperature) {
```
Comment (YurijPaharev) [inline javascript/javascript1/week2/weather.js]: You should probably add something like "getClothesBasedOnTemperature"

File: javascript/javascript1/week2/weather.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  if (temperature <= 20) {
+    return "Jackets and thermals";
+  } else if (temperature > 20 && temperature < 25) {
+    return "Denim jacket and Jeans";
+  } else if (temperature >= 25 && temperature < 40) {
+    return "Short sleeve Tshirts and Pants";
+  } else if (temperature >= 40) {
+    return "cotton clothes and linen";
+  }
```
Comment (YurijPaharev) [inline javascript/javascript1/week2/weather.js:14]: What if the temperature is more or less than 100 degrees?

File: javascript/javascript1/week2/fullname.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+console.log(getFullname("James", "Mary", true));
+
+// resolving the above code  for women
+
+const getFullnameW = (firstName, surName, useFormalName, gender) => {
+  if (useFormalName && gender === "male") {
+    return `Lord ${firstName} ${surName}`;
+  } else if (useFormalName && gender === "female") {
+    return `Lady ${firstName} ${surName}`;
+  } else return `${firstName} ${surName}`;
```
Comment (YurijPaharev) [inline javascript/javascript1/week2/fullname.js]: What if user does not provide us with the first and last names?

File: javascript/javascript1/week2/student.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+const class07Students = [];
+
+function addStudentToClass(studentName) {
+  //call function to get the strength of class
+  let strength = getNumberOfStudents(class07Students);
+  //console.log("strength of class is " + strength);
+
+  if (strength >= 7 && studentName.toUpperCase() != "QUEEN") {
```
Comment (YurijPaharev) [inline javascript/javascript1/week2/student.js]: Please use !== instead of !=

File: javascript/javascript1/week2/student.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  } else if (class07Students.includes(studentName.toUpperCase())) {
+    return `Student ${studentName} is already in the class`;
+  } else if (!studentName) {
+    return "cannot add empty string";
+  } else if (studentName.toUpperCase() === "QUEEN") {
+    class07Students.push(studentName.toUpperCase());
+    return class07Students;
+  } else if (strength < 7) {
+    class07Students.push(studentName.toUpperCase());
+    return class07Students;
```
Comment (YurijPaharev) [inline javascript/javascript1/week2/student.js]: Can you combine these two conditions?

File: javascript/javascript1/week2/student.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  }
+}
+function getNumberOfStudents(class07Students) {
+  let strength = class07Students.length;
```
Comment (YurijPaharev) [inline javascript/javascript1/week2/student.js]: Maybe you should change work "strength" to the word "capacity"

File: javascript/javascript1/week2/candystore.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+function addCandy(candyType, weight) {
+  if (candyType === "sweet") {
+    priceOfCandy = weight * candyPriceTable.sweet;
+    console.log("sweet" + priceOfCandy);
+  } else if (candyType === "chocolate") {
+    priceOfCandy = weight * candyPriceTable.chocolate;
+  } else if (candyType === "toffee") {
+    priceOfCandy = weight * candyPriceTable.toffee;
+  } else if (candyType === "chewing-gum") {
+    priceOfCandy = weight * candyPriceTable.toffee;
```
Comment (YurijPaharev) [inline javascript/javascript1/week2/candystore.js]: it is a wrong candy type you are using here

File: javascript/javascript1/week2/candystore.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+}
+function canBuyMoreCandy(boughtCandyPrices) {
+  console.log(`The total price for candy purchased ${totalPrice}`);
+  const amountToSpend = Math.floor(Math.random() * 100);
+  console.log(`Amount your hand is ${amountToSpend}`);
+  if (totalPrice < amountToSpend) {
+    return true;
+  } else {
+    return false;
+  }
```
Comment (YurijPaharev) [inline javascript/javascript1/week2/candystore.js]: Ca you rewrite it with a ternary operator?

File: javascript/javascript1/week2/candystore.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  } else if (candyType === "chewing-gum") {
+    priceOfCandy = weight * candyPriceTable.toffee;
+  }
+  boughtCandyPrices.push(priceOfCandy);
+
+  for (let i = 0; i < boughtCandyPrices.length; i++) {
+    totalPrice = totalPrice + priceOfCandy;
+  }
+
+  return totalPrice;
```
Comment (YurijPaharev) [inline javascript/javascript1/week2/candystore.js]: I would suggest making another function for calculating specifically the total price

**Review by YurijPaharev — CHANGES_REQUESTED** [review]
Very nice! Please go through my comments i order to improve some parts of your PR

File: javascript/javascript1/week2/weather.js (hunk 1 above)
Comment (RemyaManoharan) [reply javascript/javascript1/week2/weather.js]: ok

File: javascript/javascript1/week2/weather.js (hunk 2 above)
Comment (RemyaManoharan) [reply javascript/javascript1/week2/weather.js:14]: ok..I will add that condition also

File: javascript/javascript1/week2/student.js (hunk 4 above)
Comment (RemyaManoharan) [reply javascript/javascript1/week2/student.js]: ok

File: javascript/javascript1/week2/student.js (hunk 5 above)
Comment (RemyaManoharan) [reply javascript/javascript1/week2/student.js]: yes..will do

File: javascript/javascript1/week2/student.js (hunk 6 above)
Comment (RemyaManoharan) [reply javascript/javascript1/week2/student.js]: ok

File: javascript/javascript1/week2/candystore.js (hunk 7 above)
Comment (RemyaManoharan) [reply javascript/javascript1/week2/candystore.js]: sorry..when copy/paste mistake

File: javascript/javascript1/week2/candystore.js (hunk 8 above)
Comment (RemyaManoharan) [reply javascript/javascript1/week2/candystore.js]: yeah..sure

**Review by YurijPaharev — APPROVED** [review]

**Comment by RemyaManoharan** [comment]
Thanks for the valuable feed back. And your support means a lot

**Pull request merged**